### PR TITLE
Patch add gitlab support

### PR DESCRIPTION
**Pull request by AtrisMio**
Add gitlab support and refactor code.

### CODE REVIEW

**Comment by AtrisMio** [comment]
#23

**Comment by AtrisMio** [comment]
#3

**Comment by Turbo87** [comment]
@AtrisMio hey 👋  thanks for working on this! I'm currently on vacation and will review this properly once I get back.

**Comment by AtrisMio** [comment]
@Turbo87 thx for that.

**Comment by my9074** [comment]
@AtrisMio  Hey~ It's good ~~ Does this PR support private gitlab

**Comment by AtrisMio** [comment]
@my9074 Yes, It can use environment of GITLAB_SERVER.
```shell
GITLAB_SERVER=https://gitlab.my-private-site.com GITLAB_AUTH=xxxxx lerna-changelog
```

**Comment by Turbo87** [comment]
@AtrisMio this seems to be a pretty large refactoring. I'm not sure how to properly review this. Can you break this down into smaller chunks and explain them separately? 🤔

**Comment by AtrisMio** [comment]
@Turbo87 thx for reply, I will break this down in the near future.

**Comment by koka0012** [comment]
Any update?

File: src/apis/abstract-api.ts [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+  private REPO_STORED?: string;
+
+  constructor(private readonly config: Options) {
+    this.REPO_STORED = config.repo;
+
+    this.cacheDir = config.cacheDir && path.join(config.rootPath, config.cacheDir, "github");
+    this.ignoreCommitters = config.ignoreCommitters;
+    this.auth = this.getAuthToken();
+    if (!this.auth) {
+      throw new ConfigurationError("Must provide GITHUB_AUTH");
```
Comment (dbendaou) [inline src/apis/abstract-api.ts]: or `GITLAB_AUTH` if `--git-provider=gitlab` ?

**Pull request closed without merging**